lib/matplotlib/contour.py

```diff
@@ -1183,6 +1183,9 @@ def _contour_level_args(self, z, args):
         if self.filled and len(self.levels) < 2:
             raise ValueError("Filled contours require at least 2 levels.")
 
+        # Sort the contour levels.
+        if len(self.levels) > 1:
+            self.levels = list(sorted(set(self.levels)))
```

Comment:
1. Looks like the check immediately below becomes redundant.
2. The check for `len(self.levels) > 1` is not needed (it's cheap enough to sort a size-1 list).
3. Do we actually need to uniquify the levels?  I would argue that if I pass [0, 0] as input I actually want the zero-level to be drawn twice ("do what I ask for, don't guess").